include/aspect/particle/generator/interface.h

```diff
@@ -126,6 +126,21 @@ namespace aspect
           void
           parse_parameters (ParameterHandler &prm);
 
+          void
+          parse_global_parameters (ParameterHandler &prm);
+
+          bool
+          isTimeToGenerateParticles(unsigned int timeStepNumber);
+
+          bool
+          getGenerateParticlesInitially();
+
+          unsigned int
+          getParticleIdx();
```

Comment:
This is in principle a good property to know about. However I have the feeling that it somehow duplicates information that is already stored in World::next_free_particle_index. Could you take a look and think about if it is possible to remove this duplication? Maybe there is not, but I am not sure, and storing duplicated information always leads to trouble.